test/System/Err.cpp

```diff
@@ -0,0 +1,38 @@
+#include <SFML/System/Err.hpp>
+#include <sstream>
+
+#include <doctest.h>
+
+TEST_CASE("sf::err - [system]")
+{
+    SUBCASE("Overflow default buffer")
+    {
+        sf::err() << "SFML is a simple, fast, cross-platform and object-oriented multimedia API."
+            "It provides access to windowing, graphics, audio and network."
+            "It is written in C++, and has bindings for various languages such as C, .Net, Ruby, Python.";
+    }
```

Comment:
What does this test? There's no assertion 🤔

Reply:
Yeah that's a weird subcase. It's there to make sure [`DefaultErrStreamBuffer::overflow`](https://github.com/SFML/SFML/blob/master/src/SFML/System/Err.cpp#L61-L79) gets run. In theory that function could cause a crash which would be observable as a test failure but there is seemingly nothing else we can observe about `overflow`.

Reply:
Can you redirect `std::cerr` (not `sf::err()`) to at least check that the content is properly outputted?

Reply:
I tried that and it didn't work. `sf::err` and `std::cerr` are separate output streams which appears to be in the intention behind `DefaultErrStreamBuffer`'s existence. If we just wanted to use `std::cerr` then there would no longer be a reason for `sf::err` to exist.

Reply:
Yeah, it doesn't exactly [look simple](https://stackoverflow.com/a/5419409/1034248), so I'd rather not hack anything together for that. The "issue" is, that SFML directly writes to `stderr` and doesn't use `std::cerr` which could be redirected, which kind of begs the question, why we aren't using `std::cerr` to begin with. Like you can just as easily do `std::cerr.rdbuf(buffer.rdbuf())`.

Anyways off topic in this review comment.

I'd probably add a comment explaining why there's no `CHECK` where the check would usually be.

Reply:
> I'd probably add a comment explaining why there's no CHECK where the check would usually be.

Done